packages/itwinui-react/src/core/SearchBox/SearchBox.tsx

```diff
@@ -0,0 +1,398 @@
+            </>
+          ))}
+        {expandable && (
+          <>
+            <div className='iui-searchbox-collapsed'>
+              {collapsedState ?? <SearchBoxOpenButton {...openButtonProps} />}
+            </div>
+            <div className='iui-searchbox-expanded'>
+              {children ?? (
+                <>
+                  <SearchBoxInput {...inputProps} />
+                  <SearchBoxCloseButton {...closeButtonProps} />
+                </>
+              )}
+            </div>
+          </>
+        )}
+      </InputFlexContainer>
+    </SearchBoxContext.Provider>
+  );
+}) as PolymorphicForwardRefComponent<
+  'div',
+  SearchBoxOwnProps & InputFlexContainerProps
+>;
+
+export type SearchBoxProps = PolymorphicComponentProps<
+  'div',
+  SearchBoxOwnProps & InputFlexContainerProps
+>;
+
+// ----------------------------------------------------------------------------
+export type SearchBoxIconProps = PolymorphicComponentProps<'span', IconProps>;
+
+const SearchBoxIcon = React.forwardRef((props, ref) => {
+  const { size, className, children, ...rest } = props;
+  const { size: sizeContext } = useSafeContext(SearchBoxContext);
+
+  return (
+    <Icon
+      aria-hidden
+      className={cx('iui-svg-icon', 'iui-search-icon', className)}
+      size={size ?? sizeContext}
+      ref={ref}
+      {...rest}
+    >
+      {children ?? <SvgSearch />}
+    </Icon>
+  );
+}) as PolymorphicForwardRefComponent<'span', IconProps>;
+
+// ----------------------------------------------------------------------------
+
+export type SearchBoxInputProps = {
+  label?: string;
+} & React.ComponentProps<'input'>;
+
+const SearchBoxInput = React.forwardRef(
+  (props: SearchBoxInputProps, ref: React.Ref<HTMLInputElement>) => {
+    const {
+      className,
+      label = 'Search input',
+      id: idProp,
+      disabled: disabledProp,
+      ...rest
+    } = props;
+
+    const { inputId, setInputId, disabled, inputRef } =
+      useSafeContext(SearchBoxContext);
+
+    if (idProp && idProp !== inputId) {
+      setInputId(idProp);
+    }
+
+    return (
+      <input
+        id={idProp ?? inputId}
+        aria-label={label}
+        ref={useMergedRefs(ref, inputRef)}
+        type='text'
+        className={cx('iui-search-input', className)}
+        disabled={disabledProp ?? disabled}
+        {...rest}
+      />
+    );
+  },
+);
+
+// ----------------------------------------------------------------------------
+
+/**
+ * SearchBox.Button component to add to your SearchBox.
+ */
+export type SearchBoxButtonProps = PolymorphicComponentProps<
+  'button',
+  IconButtonProps
+>;
+
+const SearchBoxButton = React.forwardRef((props, ref) => {
+  const { children, size, disabled: disabledProp, ...rest } = props;
+  const { size: sizeContext, disabled } = useSafeContext(SearchBoxContext);
+
+  return (
+    <IconButton
+      styleType='borderless'
+      size={size ?? sizeContext}
+      ref={ref}
+      disabled={disabledProp ?? disabled}
```

Comment:
i think you don't even need to destructure the `size` and `disabled` props, because `...rest` will override the first instance you set from context
![](https://user-images.githubusercontent.com/9084735/235931418-cc86e9ca-5593-4f65-860a-2484a9acc04d.png)

same comment in other subcomponents

Reply:
Removed size and disabled props.